lib/src/services/jira_api.test.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable functional/prefer-immutable-types */
+import * as E from "fp-ts/lib/Either";
+import JiraApi from "jira-client";
+import { JiraClient, jiraClient } from "./jira_api";
+import { boardReturn, searchJiraReturn } from "./test_data/jira_api_data";
+import { readonlyDate } from "readonly-types";
+import { pipe } from "fp-ts/lib/function";
+
+jest.mock("jira-client", () => {
+  return jest.fn().mockImplementation(() => {
+    return {
+      searchJira: jest
+        .fn()
+        .mockImplementation(() =>
+          Promise.resolve({ issues: searchJiraReturn })
+        ),
```

Comment:
Instead of mocking the implementation and returning a Promise, see if you can accomplish what you need with `mockResolvedValue` instead.